### PR TITLE
flamenco: ALUT resolution pre-checks

### PR DESCRIPTION
**Pull request by mjain-jump**

### CODE REVIEW

File: src/flamenco/runtime/fd_executor.c [hunk 1]
```diff
@@ -456,9 +456,14 @@ fd_executor_setup_accessed_accounts_for_txn( fd_exec_txn_ctx_t * txn_ctx ) {
   txn_ctx->accounts_cnt += (uchar) txn_ctx->txn_descriptor->acct_addr_cnt;
 
   if( txn_ctx->txn_descriptor->transaction_version == FD_TXN_V0 ) {
+    /* https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/runtime/src/bank/address_lookup_table.rs#L44-L48 */
+    fd_slot_hashes_t const * slot_hashes = fd_sysvar_cache_slot_hashes( txn_ctx->slot_ctx->sysvar_cache );
+    if( FD_UNLIKELY( !slot_hashes ) ) {
+      return FD_RUNTIME_TXN_ERR_ACCOUNT_NOT_FOUND;
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/fd_executor.c:462]: this error seems to be a AddressLoaderError::SlotHashesSysvarNotFound in the attached permalink

File: src/flamenco/runtime/fd_runtime.c [hunk 2]
```diff
@@ -777,15 +778,15 @@ fd_runtime_prepare_txns_start( fd_exec_slot_ctx_t *         slot_ctx,
 
     fd_rawtxn_b_t raw_txn = { .raw = txn->payload, .txn_sz = (ushort)txn->payload_sz };
 
-    int res = fd_execute_txn_prepare_start( slot_ctx, txn_ctx, txn_descriptor, &raw_txn );
-    if( res ) {
-      task_info[txn_idx].exec_res = res;
+    int err = fd_execute_txn_prepare_start( slot_ctx, txn_ctx, txn_descriptor, &raw_txn );
+    if( err ) {
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/fd_runtime.c:782]: prob can make this an fd_unlikely while youre at it

File: src/flamenco/runtime/program/fd_address_lookup_table_program.h [hunk 3]
```diff
@@ -15,6 +15,17 @@ FD_PROTOTYPES_BEGIN
...
+/**********************************************************************/
+/* Public API                                                         */
+/**********************************************************************/
+
+int
+fd_get_active_addresses_len( fd_address_lookup_table_t *       self,
+                             ulong                             current_slot,
+                             fd_slot_hash_t const *            slot_hashes,
+                             ulong                             addresses_len,
+                             ulong *                           active_addresses_len );
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/program/fd_address_lookup_table_program.h:27]: ultra mega nit: dont need that much spacing between variables and their types

File: src/flamenco/runtime/program/fd_address_lookup_table_program.c [hunk 4]
```diff
@@ -989,3 +967,94 @@ fd_address_lookup_table_program_execute( fd_exec_instr_ctx_t * ctx ) {
 
   return FD_EXECUTOR_INSTR_SUCCESS;
 }
+
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/program/fd_address_lookup_table_program.c:970]: can you fix vertical alignment and use LU as needed

File: src/flamenco/runtime/program/fd_address_lookup_table_program.c [hunk 5]
```diff
@@ -989,3 +967,94 @@ fd_address_lookup_table_program_execute( fd_exec_instr_ctx_t * ctx ) {
...
+/**********************************************************************/
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/program/fd_address_lookup_table_program.c:971]: For all the permalinks don't use the agave fork urls

File: src/flamenco/runtime/program/fd_address_lookup_table_program.c [hunk 6]
```diff
@@ -989,3 +967,94 @@ fd_address_lookup_table_program_execute( fd_exec_instr_ctx_t * ctx ) {
...
+/* Sets active_addresses_len on success
+   https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/sdk/program/src/address_lookup_table/state.rs#L142-L164 */
+int
+fd_get_active_addresses_len( fd_address_lookup_table_t *       self,
+                             ulong                             current_slot,
+                             fd_slot_hash_t const *            slot_hashes,
+                             ulong                             addresses_len,
+                             ulong *                           active_addresses_len ) {
+  /* https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/sdk/program/src/address_lookup_table/state.rs#L147-L152 */
+  if( !is_active( self, current_slot, slot_hashes ) ) {
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/program/fd_address_lookup_table_program.c:1050]: FD_UNLIKELY

File: src/flamenco/runtime/program/fd_address_lookup_table_program.c [hunk 7]
```diff
@@ -989,3 +967,94 @@ fd_address_lookup_table_program_execute( fd_exec_instr_ctx_t * ctx ) {
...
+    return FD_ADDRLUT_STATUS_DEACTIVATING;
+  }
+
+  /* https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/sdk/program/src/address_lookup_table/state.rs#L102 */
+  return FD_ADDRLUT_STATUS_DEACTIVATED;
+}
+
+/* https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/sdk/program/src/address_lookup_table/state.rs#L72-L78 */
+static uchar
+is_active( fd_address_lookup_table_t const *       self,
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/program/fd_address_lookup_table_program.c:1026]: spaces

File: src/flamenco/runtime/program/fd_address_lookup_table_program.c [hunk 8]
```diff
@@ -989,3 +967,94 @@ fd_address_lookup_table_program_execute( fd_exec_instr_ctx_t * ctx ) {
...
+           ulong                                   current_slot,
+           fd_slot_hash_t const *                  slot_hashes ) {
+  /* https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/sdk/program/src/address_lookup_table/state.rs#L73-L77*/
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/program/fd_address_lookup_table_program.c:1029]: space at end of this line

**Review by ibhatt-jumptrading — CHANGES_REQUESTED** [review]
mostly lgtm, style nits that you can ignore

File: src/flamenco/runtime/fd_executor.c (hunk 1 above)
Comment (mjain-jump) [reply src/flamenco/runtime/fd_executor.c:462]: In error conversion, it gets converted into the above transaction error

**Review by ibhatt-jumptrading — APPROVED** [review]

**Pull request merged**